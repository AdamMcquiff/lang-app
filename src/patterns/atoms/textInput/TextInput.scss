.a-text-input {
    background-color: $dark-blue;
}