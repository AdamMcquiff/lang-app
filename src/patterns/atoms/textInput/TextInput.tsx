import * as React from 'react';
import './TextInput.scss';

interface ITextInputProps {
    textLabel: string,
    a11yLabel: string,
    onTextChange?: (e: any) => void
}

class TextInput extends React.Component<ITextInputProps> {
    public static defaultProps = {
        a11yLabel: 'Give me a label',
        onClick: () => null,
        textLabel: 'Give me a label',
    }

    public render() {
        return (
            <label className="a-text-input">
                <span className="a-text-input__a11y-label"
                      hidden={true}>
                    {this.props.a11yLabel}
                </span>
                
                <span className="a-text-input__text-label"
                      aria-hidden={true}>
                    {this.props.textLabel}
                </span>

                <input className="a-text-input__field"
                       onChange={this.props.onTextChange}
                       type="text" />
            </label>
        )
    }
}

export default TextInput;