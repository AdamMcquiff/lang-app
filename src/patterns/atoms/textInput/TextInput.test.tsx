import { mount, shallow } from 'enzyme';
import * as React from 'react';
import TextInput from './TextInput';

describe('TextInput', () => {
    const props = {
        a11yLabel: "Some label",
        onTextChange: jest.fn(),
        textLabel: "Some label"
    };

    it('renders without crashing with no props', () => {
        const component = shallow(<TextInput />);
        expect(component).toMatchSnapshot();
    });

    it('renders without crashing with all props', () => {
        const component = shallow(<TextInput {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly with a11yLabel prop passed', () => {
        const component = mount(<TextInput a11yLabel={props.a11yLabel} />);
        const renderedA11yLabel = component.find('.a-text-input__a11y-label').text();
        expect(renderedA11yLabel).toEqual(props.a11yLabel); 
    });

    it('renders correctly with textLabel prop passed', () => {
        const component = mount(<TextInput textLabel={props.textLabel} />);
        const renderedTextLabel = component.find('.a-text-input__text-label').text();
        expect(renderedTextLabel).toEqual(props.textLabel); 
    });

    it('renders correctly with onTextChange prop passed', () => {
        const component = mount(<TextInput onTextChange={props.onTextChange} />);
        component.find('input').simulate('change');
        expect(props.onTextChange).toBeCalled(); 
    });
});