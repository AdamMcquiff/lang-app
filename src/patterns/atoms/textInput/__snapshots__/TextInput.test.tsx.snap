// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders without crashing 1`] = `
<label
  className="a-text-input"
>
  <span
    className="a-text-input__a11y-label"
    hidden={true}
  >
    Give me a label
  </span>
  <span
    aria-hidden={true}
    className="a-text-input__text-label"
  >
    Give me a label
  </span>
  <input
    className="a-text-input__field"
    type="text"
  />
</label>
`;
